components/buttons/RemoveUserButton/RemoveUserButton.js

```diff
@@ -0,0 +1,53 @@
+import IconButton from '@material-ui/core/IconButton'
+import CircularProgress from '@material-ui/core/CircularProgress'
+import DeleteIcon from '@material-ui/icons/Delete'
+import axios from 'axios'
+import { useState } from 'react'
+
+const url = (uid) => `/api/friends/removeFriend/${uid}`
+
+export default function RemoveUserButton(props) {
+	const { onRemoveFriend, friend } = props
+
+	const [request, setRequest] = useState({
+		loading: false,
+		error: false,
+	})
+
+	const onClick = async () => {
```

Comment:
This post should be in redux as well. If this onClick is doing more than one state update, then it should be an imported helper function that is tested to ensure the dispatches to state are being called